tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/contact/EmailAddressContact.scala

```diff
@@ -79,6 +82,26 @@ class InMemoryEmailAddressContactSearchEngine extends EmailAddressContactSearchE
     SMono.fromCallable(() => domainList.put(domain, addressContact))
       .`then`(SMono.just(addressContact))
 
+  override def delete(accountId: AccountId, mailAddress: MailAddress): Publisher[Void] =
+    getContact(accountId, mailAddress)
+      .map(contact => emailList.remove(accountId, contact))
+      .`then`()
+
+  private def getContact(accountId: AccountId, mailAddress: MailAddress): SMono[EmailAddressContact] =
+    SFlux.fromIterable(emailList.get(accountId).asScala)
+      .filter(_.fields.address.equals(mailAddress))
+      .singleOrEmpty()
+
+  override def delete(domain: Domain, mailAddress: MailAddress): Publisher[Void] =
+    getDomainContact(domain, mailAddress)
+      .map(contact => domainList.remove(domain, contact))
+      .`then`()
+
+  private def getDomainContact(domain: Domain, mailAddress: MailAddress): SMono[EmailAddressContact] =
+    SFlux.fromIterable(domainList.get(domain).asScala)
+      .filter(_.fields.address.equals(mailAddress))
+      .singleOrEmpty()
```

Comment:
I think should we delete all the records (for idempotent).

FYI on the ES side, we have a unique ID for the document (domain + mail address) => reduce duplicated index data, just need to delete one record :') For memory implementation, I think we can use Guava Table, however not really strict though.

Reply:
> I think should we delete all the records (for idempotent).

Hmmm why? I want to delete one contact for a domain, not all?

> For memory implementation, I think we can use Guava Table, however not really strict though.

No need I think

Reply:
> Hmmm why? I want to delete one contact for a domain, not all?

Is there any case where we have many contacts with the same mail address in the real world?

Reply:
Usually when you create a mail address at any mail provider, I believe if you try to choose a mail address that is already used by someone else, you get an error. For the good reason that if your server gets a mail addressed to bob@linagora.com but you have Bob and Alice linked to it, to whom do you deliver it? Mail addresses for a domain are unique I believe

Reply:
> Mail addresses for a domain are supposed to be unique I believe

So that is why contact for mail address should be unique also. With current memory implementation, we could index many contacts with the same mail address => deleting just one contact is not enough.

This is what I mean :)

Reply:
ah didn't get your point that way... makes sense :) not making the task easier but all right, I give you credit for this one ;)

Reply:
Yeh, that is it. Let me handle it if you need some help :p 
For the memory implementation, you can have a look at `MemoryCustomIdentityDAO`.
I think we can create another ticket for implementing ES6/ES7 for these update/contact APIs if necessary?

Reply:
As you wish, I'm banging my head at trying to merge latest james in tmail for now :D

Reply:
Agree for indempotance.